src/renderer/qss/js/qssFocusManager.js

```diff
@@ -0,0 +1,343 @@
+/**
+ * Defines a `focusManager` component for the QSS
+ *
+ * A focus manager which enables navigation between the various focusable elements
+ * using (Shift +) Tab and all arrow keys.
+ * Copyright 2017 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ * The research leading to these results has received funding from the European Union's
+ * Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+/* global fluid jQuery */
+
+"use strict";
+(function (fluid, jQuery) {
+    var gpii = fluid.registerNamespace("gpii");
+
+    /**
+     * A special type of focusManager which enables navigation between the focusable
+     * elements using Tab, Shift + Tab, Arrow Up, Arrow Down, Arrow Left and the
+     * Arrow Right key. The navigation focus basically moves in the visual direction.
+     */
+    fluid.defaults("gpii.qss.qssFocusManager", {
+        gradeNames: ["gpii.qss.horizontalFocusManager", "gpii.qss.verticalFocusManager"],
+
+        maxElementsPerFocusGroup: 2,
+
+        styles: {
+            button: "fl-qss-button",
+            smallButton: "fl-qss-smallButton"
+        },
+
+        listeners: {
+            "onArrowUpPressed.impl": {
```

Comment:
These should be rewritten to visibly make calls to the focusedNearestHorizontally etc. methods